docs/content/guides/dagster-pipes/integrating-kubernetes-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,242 @@
+
+**Note**: Execution can take several minutes even for trivial scripts due to cluster provisioning times.
+
+<!--
+TODO: Takeout if not applicable
+
+When the job finishes executing, Dagster Pipes will return a <PyObject module="dagster_pipes" object="PipesClientCompletedInvocation" /> object. This object exposes a `get_results` method, which you can use to access the results reported by Kubernetes.  -->
+
+### Step 1.2: Define the Kubernetes Pipes resource
+
+The [`dagster-k8s`](/\_apidocs/libraries/dagster-Kubernetes) library provides a <PyObject module="dagster_k8s" object="PipesK8sClient" />, which is a pre-built Dagster resource that allows you to quickly get Pipes working with your Kubernetes workspace.
+
+1. In the imports section of `dagster_k8s_pipes.py`, add the following:
+
+   ```python
+   # dagster_k8s_pipes.py
+
+   import kubernetes
+   ```
+
+2. Next, copy and paste the following to the bottom of `dagster_k8s_pipes.py`:
+
+   ```python
+   # dagster_k8s_pipes.py
+
+   kubernetes.config.load_kube_config(
+     config_file=os.path.expanduser("~/.kube/config"),
+     context="your_context_name",
+   )
+   ```
+
+   In this example, we the `kubernetes` Python client to configure the Kubernetes pod. The `load_kube_config` method loads authentication information from the provided kube-config (`config_file`) file and sets the active context (`context`).
+
+### Step 1.3: Update the Definitions object
+
+To wrap up this section, you’ll add the asset and Kubernetes resource to your project’s code location via the <PyObject object="Definitions" /> object. This makes the resource available to other Dagster definitions in the project.
+
+Copy and paste the following to the bottom of `dagster_k8s_pipes.py`:
+
+```python
+# dagster_k8s_pipes.py
+
+defs = Definitions(
+  assets=[k8s_pipes_asset],
+  resources={
+    "k8s_pipes_client": PipesK8sClient(),
+  },
+)
+```
+
+At this point, `dagster_k8s_pipes.py` should look like the following:
+
+```python
+# dagster_k8s_pipes.py
+
+from dagster import AssetExecutionContext, Definitions, asset
+from dagster_k8s import PipesK8sClient
+import kubernetes
+
+
+@asset
+def k8s_pipes_asset(context: AssetExecutionContext, k8s_pipes_client: PipesK8sClient):
+  return k8s_pipes_client.run(
+    context=context,
+    image="pipes-materialize:latest",
+  ).get_materialize_result()
+
+
+kubernetes.config.load_kube_config(
+  config_file=os.path.expanduser("~/.kube/config"),
+  context="your_context_name",
+)
+
+
+defs = Definitions(
+  assets=[k8s_pipes_asset],
+  resources={
+    "k8s_pipes_client": PipesK8sClient(),
+  },
+)
+```
+
+---
+
+## Step 2: Modify the Kubernetes code
+
+The next step is to modify the code in Kubernetes to work with Dagster Pipes.
+
+1. In the imports section of the file, add the following:
+
+   ```python
+   # existing_kubernetes_code.py
+
+   from dagster_pipes import TODO, TODO, open_dagster_pipes
```

Comment:
In the Databricks version of this guide (which is admittedly still a WIP), we specified the context injector/loader and message reader to use ([link here](https://github.com/dagster-io/dagster/pull/17114/files#diff-dd1c3d7589d4c49b377c425f01dac75a6e12d735ec3f9be490fe334721b1cd88R234)). What should K8s be using?

Reply:
the defaults work for k8s databricks is a special case where you need to specify those